corehq/apps/reports/standard/users/reports.py

```diff
@@ -179,15 +180,39 @@ def _html_list(changes, unstyled=True):
 
     properties = UserHistoryReport.get_primary_properties(domain)
     properties.pop("user_data", None)
-    primary_changes = {
-        properties.get(key, key): value for key, value in changes.items()
-        if key in properties
-    }
-    more_count = len(changes) - len(primary_changes)
+    primary_changes = {}
+
+    for key, value in changes.items():
+        if key in properties:
+            if key == 'commcare_location_id':
+                primary_changes[_("primary location (mobile users only)")] = cached_location_id_to_name(value)
+            if key == 'location_id':
+                primary_changes["location"] = cached_location_id_to_name(value)
+            else:
+                primary_changes[key] = value
+
+    all_changes = {}
+
+    for key, value in changes.items():
+        if key == 'user_data':
```

Comment:
I see you anticipate `commcare_location_id` and `location_id` in both loops, but I believe the former only exists inside `user_data` (aka metadata), and the latter only exists outside of it.